platform/ios/src/MGLMapView.mm

```diff
@@ -4688,15 +4691,30 @@ - (void)applyIconIdentifier:(NSString *)iconIdentifier toAnnotationsWithImageReu
 
 #pragma mark - User Location -
 
+- (void)setLocationManager:(id<MGLLocationManager>)locationManager
+{
+    if (!locationManager) {
+        [self.locationManager stopUpdatingLocation];
+        [self.locationManager stopUpdatingHeading];
+        self.locationManager.delegate = nil;
```

Comment:
We should set the delegate to nil if `locationManager != self.locationManager`.

Reply:
Unlike `CLLocationManager` our delegate is marked as `weak` but I think make sense cleanup before assigning the new delegate.

Reply:
For clarity, I meant the `self.locationManager.delegate = nil;` line